pipeline-runner/R/gem2s-2-load_cellranger_files.R

```diff
@@ -98,3 +99,68 @@ format_annot <- function(annot_list) {
   rownames(annot) <- annot$input
   return(annot)
 }
+
+
+read_rhapsody_matrix <- function(config, input_dir) {
+  counts_list <- list()
+  annot_list <- list()
+
+  samples <- config$samples
+  message("Samples to include in the analysis:\n- ", paste(samples, collapse = "\n- "))
+  message("Loading rhapsody data set from input folder.")
+
+
+  for (sample in samples) {
+    sample_dir <- file.path(input_dir, sample)
+    sample_fpaths <- file.path(sample_dir, "expression_matrix.st")
+
+    message("\nSample --> ", sample)
+    message("Reading files from ", sample_dir, " --> ", paste(sample_fpaths, collapse = " - "))
+
+
+    counts <- data.table::fread(sample_fpaths)
+
+    # catch absent DBEC column
+    if ("DBEC_Adjusted_Molecules" %in% names(counts)) {
+      counts <- counts[, c("Cell_Index", "Gene", "DBEC_Adjusted_Molecules")]
+    } else {
+      counts <- counts[, c("Cell_Index", "Gene", "RSEC_Adjusted_Molecules")]
+    }
+
+    # order by cell indices and gene, to ensure correct cell_index_j to
+    # column name association when using frank
+    data.table::setorder(counts, Cell_Index, Gene)
+
+    counts[, Gene := factor(Gene)]
+    counts[, gene_i := as.integer(Gene)]
+
+    # to create small sparse matrix, and retain original cell indices ("barcodes")
+    counts[, cell_index_j := data.table::frank(counts[, Cell_Index], ties.method = "dense")]
+
+    # create row and column names
+    dimnames_i <- levels(counts[, Gene])
+    dimnames_j <- unique(counts[, Cell_Index])
+
+    counts <- Matrix::sparseMatrix(
+      i = counts[, gene_i],
+      j = counts[, cell_index_j],
+      x = counts[, DBEC_Adjusted_Molecules],
+      dimnames = list(dimnames_i, dimnames_j)
+    )
```

Comment:
```suggestion
    # gene indices
    counts[, gene_i := as.integer(factor(Gene))]

    # use rank of cell indices so that absent cell ids
    counts[, cell_index_j := as.integer(factor(counts$Cell_Index))]

    counts <- Matrix::sparseMatrix(
      i = counts$gene_i,
      j = counts$cell_index_j,
      x = counts$DBEC_Adjusted_Molecules,
      dimnames = list(unique(counts$Gene), unique(counts$Cell_Index))
    )
```